config.go

```diff
@@ -514,7 +516,7 @@ func loadConfig() (*config, []string, error) {
 
 	// Create the home directory if it doesn't already exist.
 	funcName := "loadConfig"
-	err = os.MkdirAll(defaultHomeDir, 0700)
+	err = os.MkdirAll(cfg.HomeDir, 0700)
```

Comment:
What if you run `btcd` for the first time and you don't have `.btcd` (default directory for linux)?
https://github.com/btcsuite/btcd/blob/e5521de652133c1d59df6fc505fd3f6e5c23bea9/config.go#L70-L76
Everything else gonna fail, because `defaultHomeDir` directory wasn't created for these files.

Also, I'm not sure about the behaviour of other systems. Is it really safe to use `cfg.HomeDir` as an empty argument  string (default case if you don't provide this flag) for Windows?